doc/design/persistent_volumes_etcd_data.md

```diff
@@ -0,0 +1,79 @@
+# Use Persistent Volumes for etcd data
+
+## Proposal and Motivation
+
+Today etcd-operator creates ephemeral etcd members: in their pod the etcd data is stored inside a volume of type emptyDir and the pod restart policy is set to `Never`.
+
+So, if a pod is deleted from the API server (manually or automatically: i.e a node goes down and the node controller, after a timeout, evicts pods marking them for termination), etcd-operator will remove the related etcd member from the etcd cluster and will create a new pod (with a new name and an empty data dir) and add that new etcd member to the etcd cluster.
+
+When a majority of pods is deleted at the same time the etcd cluster will become unquorate and there'll be no way to restore its quorum since the members and their data are gone. So the unique solution (that etcd-operator currently do) is to reinitialize a new cluster restoring its data from a backup (if available).
+
+This proposal is to fix such case achieving a persistent/durable etcd cluster.
+
+## Proposed Changes
+
+#### Changes
+
+Add an option to enable storing etcd data inside a persistent volume and related reconcile logic changes.
+
+The change to be effective and cover the above failure scenario is split in two parts.
+
+### Part 1
+
+- Add to the cluster spec PodPolicy an option to enable putting etcd data inside a Persistent Volume
+- If podPolicy has persistent volumes enabled, when creating a pod also create a Persistent Volume Claim and use it as the volume source for etcd-data
+- Handle etcd data persistent volume garbage collection like already done with other resources. While backup pv aren't garbage collected, etcd data pv needs to be garbage collected when not needed anymore.
```

Comment:
> Handle etcd data persistent volume garbage collection like already done with other resources.

This is different. Operator only creates PVC. It doesn't create PV. So I assume here PV is managed by controller manager. When PV is released, k8s controller will remove it. Please write more details about this process.
 
Add more details about how to cleanup PVC too. If a member is removed, we should also delete its PVC.

> While backup pv aren't garbage collected

Why is it related to backup PV?

Reply:
forgot to add `claim` fixed and detailed.


> Why is it related to backup PV?

just to be clear that only etcd data PVCs will be GCed, backups PVC won't be touched by this proposal. Will remove it if it's tacit.